api/server/server.go

```diff
@@ -137,6 +139,12 @@ func startServer(name string, sockBase string, port uint16, routes []*Route) err
 	for _, v := range routes {
 		router.Methods(v.verb).Path(v.path).HandlerFunc(v.fn)
 	}
+	//Add only to cluster routes
+	clusterApi := newClusterAPI()
+	sm := clusterApi.RegisterSecretManager(fake.New())
+	for _, v := range sm.Routes() {
```

Comment:
How to add append secrets.Route() into clusterAPI.Routes()?

Reply:
Ah, I see the issue. Routes are not a property of the SecretManager(), they are a property of the server. The SecretManager is an implementation of the handlers to these routes, but the routes are not a property of the SecretManager. For this reason, the `routes` in the arguments should have been filled in already by the caller.

Reply:
so should I move routes back into api/cluster/routes.go like this 
```{verb: "GET", path: clusterVersion("checksecretslogin", cluster.APIVersion), fn: c.secretLoginCheck},```
->https://github.com/ram-infrac/openstorage/commit/4078f23de524b46b22ef314df58bb20a1a6d05fe#diff-17b630a1004c86897f2252802622cf29R34.